readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/writer/FileSystemUtils.java

```diff
@@ -0,0 +1,29 @@
+package com.criteo.hadoop.garmadon.hdfs.writer;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.util.Collection;
+
+public final class FileSystemUtils {
+    private FileSystemUtils() {
+    }
+
+    /**
+     * Ensure paths exist and are directories. Otherwise create them.
+     *
+     * @param dirs          Directories that need to exist
+     * @param fs            Filesystem to which these directories should belong
+     * @throws IOException  When failing to create any of the directories
+     */
+    public static boolean ensureDirectoriesExist(Collection<Path> dirs, FileSystem fs) throws IOException {
+        for (Path dir : dirs) {
+            if (!fs.exists(dir) && !fs.mkdirs(dir)) {
+                return false;
```

Comment:
why not throwing an IOException (or other type of exception) if you fail to create the dir?
Usage seems to abort current the process anyway. Exception message can include the right infos

Reply:
for the log warning you can catch the IOException and log the message

Reply:
Anything is possible, question is: is that really an issue? :-)

Reply:
My point is to merge the usages.
Only dealing with Exception here seems the good way as this will be exceptional...